### PR TITLE
Add cffi support to @cfunc

### PR DESCRIPTION
**Pull request by pitrou**

### CODE REVIEW

File: docs/source/reference/jit-compilation.rst [hunk 1]
```diff
@@ -405,6 +405,11 @@ C callbacks
 
       The address of the compiled C callback, as an integer.
 
+   .. attribute:: cffi
+
+      A `cffi`_ function pointer instance, suitable for passing as an
+      argument to `cffi`_-wrapped functions.
```
Comment (seibert) [inline docs/source/reference/jit-compilation.rst]: We should note that the function pointer is `void *` so CFFI will not be able to verify pointer compatibility with the argument type.  Someone might be expected that kind of check to happen.

**Comment by pitrou** [comment]
Ok, I've clarified the docs.

**Comment by seibert** [comment]
Looks like Windows isn't happy: https://ci.appveyor.com/project/pitrou/numba/build/1.0.962#L207

**Comment by pitrou** [comment]
Ok, I fixed the Windows issue.

**Pull request merged**